container/shim/src/fetchers/lassie.js

```diff
@@ -167,11 +191,36 @@ async function getRequestedBlockFromCar(streamIn, streamOut, requestedCidV1, pat
         throw new Error(`Requested CID ${requestedCidV1} doesn't equal first block CID ${rootCid}.`);
       }
 
-      sendBlockResponse(streamOut, bytes, cidV1.toString())
+      sendBlockResponse(streamOut, bytes, cidV1.toString());
     } else {
       const cacheKey = cidToCacheKey(cidV1);
       blockCache.set(cacheKey, bytes);
     }
     count++;
   }
 }
+
+async function queueMetricsReport() {
+  const date = lastMetricsReportDate;
+  const canReport = !date || new Date() - date > METRICS_REPORT_INTERVAL;
+  if (!canReport || !metrics.length) {
+    return;
+  }
+
+  const logsToReport = metrics.splice(0);
```

Comment:
```suggestion
  const logsToReport = [...metrics];
```

Reply:
Splice also empties the array, which needs to happen at some point.

Reply:
Gotcha. Since both Diego and me missed this, what do you think about making this more explicit instead?

Reply:
ah, slice vs splice always gets me!

Reply:
How about this?
```js
  const logsToReport = metrics;
  metrics = []
  ```

Reply:
hopefully fixed by https://github.com/filecoin-saturn/L1-node/pull/221/commits/c90c0307a1c3cd05948377c2c8a67be7a578231a?